src/combobox/index.ts

```diff
@@ -290,32 +294,54 @@ export class ComboBox extends I18nMixin(ThemedMixin(FocusMixin(WidgetBase)))<Com
 	}
 
 	protected getRootClasses(): (string | null)[] {
-		const { clearable, invalid } = this.properties;
+		const {
+			clearable
+		} = this.properties;
+
+		const {
+			valid
+		} = this.validity;
 		const focus = this.meta(Focus).get('root');
 
 		return [
```

Comment:
can you just move this into the main `render` function please?

Reply:
Done